src/providers/DatasetBasketProvider.tsx

```diff
@@ -0,0 +1,50 @@
+// SPDX-FileCopyrightText: 2024 PNED G.I.E.
+//
+// SPDX-License-Identifier: Apache-2.0
+"use client";
+import React, { createContext, useContext, useState, ReactNode } from "react";
+import { Dataset } from "@/types/dataset.types";
+
+interface DatasetBasketContextType {
+  basket: Dataset[];
+  addDatasetToBasket: (dataset: Dataset) => void;
+  removeDatasetFromBasket: (dataset: Dataset) => void;
+}
+
+const DatasetBasketContext = createContext<
+  DatasetBasketContextType | undefined
+>(undefined);
+
+export const DatasetBasketProvider = ({
+  children,
+}: {
+  children: ReactNode;
+}) => {
+  const [basket, setBasket] = useState<Dataset[]>([]);
```

Comment:
I will still have to store them in cookies to persist it. Will be a separate PR

Reply:
Cookies (server side?) or local storage?

Reply:
it has to be on browser side if we want to make add to basket as javascript based action as compared to page reload.

I was thinking to use cookies because it could work for both client side as well as server side components. 
However i need to run this to be fully sure.